packages/editor/src/extensions/link/tests/link.test.ts

```diff
@@ -0,0 +1,117 @@
+/*
+This file is part of the Notesnook project (https://notesnook.com/)
+
+Copyright (C) 2023 Streetwriters (Private) Limited
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { describe, expect, it, test } from "vitest";
+import { createEditor, h } from "../../../../test-utils";
+import { Link, linkRegex } from "../link";
+
+test("link regex", () => {
+  const input = [
+    {
+      text: "[Example](https://www.google.com)",
+      expected: ["[Example](https://www.google.com)"]
+    },
+    {
+      text: "[Example](https://www.example1.com) and [Example2](https://www.example2.com)",
+      expected: [
+        "[Example](https://www.example1.com)",
+        "[Example2](https://www.example2.com)"
+      ]
+    },
+    {
+      text: "[]()",
+      expected: null
+    },
+    {
+      text: "[] [Link](http://example.com)",
+      expected: ["[Link](http://example.com)"]
+    },
+    {
+      text: "[not a link] [Link](http://example.com)",
+      expected: ["[Link](http://example.com)"]
+    },
+    {
+      text: "[NoLink]",
+      expected: null
+    }
+  ];
+
+  input.forEach(({ text, expected }) => {
+    const result = text.match(linkRegex);
+    expect(result).toEqual(expected);
+  });
```

Comment:
Instead of looping inside the `test`, move the loop outside and the `test` in. That way we'll have multiple tests and if any one fails, we'll quickly know which one exactly failed.

Reply:
ohh, nice suggestion